cclib/bridge/cclib2pyscf.py

```diff
@@ -19,15 +23,38 @@ def _check_pyscf(found_pyscf):
         raise ImportError("You must install `pyscf` to use this function")
 
 
-def makepyscf(atomcoords, atomnos, charge=0, mult=1):
+def makepyscf(data, charge=0, mult=1):
     """Create a Pyscf Molecule."""
     _check_pyscf(_found_pyscf)
     mol = gto.Mole(
-        atom = [['{}'.format(atomnos[i]),atomcoords[i]] for i in range(len(atomcoords))],
+        atom=[
+            ["{}".format(data.atomnos[i]), data.atomcoords[-1][i]]
+            for i in range(data.natom)
+        ],
         unit="Angstrom",
         charge=charge,
-        multiplicity=mult
+        multiplicity=mult,
     )
-    return  mol
-  
+    inputattr = data.__dict__
+    if "gbasis" in inputattr:
+        basis = {}  # object for internal PySCF format
+        uatoms, uatoms_idx = np.unique(
+            data.atomnos, return_index=True
+        )  # find unique atoms
+        for idx, i in enumerate(uatoms_idx):
+            curr_atom_basis = data.gbasis[i]
+            for jdx, j in enumerate(curr_atom_basis):
+                curr_l = j[0]
+                curr_e_prim = j[1]
+                new_list = [l_sym2num["{}".format(curr_l)]]
+                new_list += curr_e_prim
+                pt = PeriodicTable()

```

Comment:
Can you put this at the top level, still inside the function but outside any loops?

Reply:
Yes, of course. Not sure what I was thinking there. This should be addressed now.